scripts/create_newcase

```diff
@@ -217,6 +217,14 @@ def _main_func(description):
     expect(not (os.path.exists(caseroot) and not test),
            "Case directory {} already exists".format(caseroot))
 
+    # create_newcase ... --test ... throws a CIMEError along with
+    # a very stern warning message to the user
+    # if it detects that it was invoked outside of create_test
+    if test:
+        expect(("FROM_CREATE_TEST" in os.environ and os.environ["FROM_CREATE_TEST"] == "True"),
```

Comment:
We only want a warning, not an error.

Reply:
JimE mentioned in the original issue that the message should be an error, so I am resolving this for now.